### PR TITLE
create spdlog service after workbench is restored

### LINKED ISSUE #89785: Logging service delays renderer
* run `code --prof-startup`
* loading the `spdlog` takes around 15-20ms 
* because the logger is initialised soo early it delays creating the window (screen shot)

I think the better approach would be use the lifecycle knowledge that @bpasero introduced to buffer log messages until the renderer has been started. Given that [`createSpdLogLogger`](https://github.com/microsoft/vscode/blob/846cff1e8fd0a3aa140341bb63e8e6b8c7074289/src/vs/platform/log/node/spdlogService.ts#L10) is already async this could be little work

Screenshot 2020-01-31 at 14 51 55

### PR DESCRIPTION
**Pull request by sandy081**
This PR fixes #89785

Delay creating the spdlogger until workbench is restored.

### CODE REVIEW

File: src/vs/workbench/electron-browser/desktop.main.ts [hunk 1]
```diff
@@ -314,26 +322,53 @@ class DesktopMain extends Disposable {
 		}
 	}
 
-	private createLogService(mainProcessService: IMainProcessService, environmentService: IWorkbenchEnvironmentService): ILogService {
+}
+
```
Comment (bpasero) [inline src/vs/workbench/electron-browser/desktop.main.ts:326]: Extra empty line?

File: src/vs/workbench/electron-browser/desktop.main.ts [hunk 2]
```diff
@@ -122,6 +124,12 @@ class DesktopMain extends Disposable {
 		// Startup
 		const instantiationService = workbench.startup();
 
+		// Lifecycle Listeners
```
Comment (bpasero) [inline src/vs/workbench/electron-browser/desktop.main.ts:127]: If we move `DesktopLogService` to `src/vs/workbench/services/log/electron-browser` we could actually simply contribute a workbench contribution from there as well that starts on that phase. Would make it nicely selfcontained.

**Review by bpasero — CHANGES_REQUESTED** [review]

**Comment by sandy081** [comment]
Good idea. Moved `DesktopLogService` to separate file.

**Review by bpasero — APPROVED** [review]
Great 👍

**Pull request merged**